### PR TITLE
Messages API JSDoc

### PR DESCRIPTION
**Pull request by ctrlaltdavid**

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
**Test Plan**

Run the \scripts\developer\tests\messagesTest.js and check that it says "received" rather than "recieved".

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8051/app_12650_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8051/HighFidelity-Beta-PR12650-63e07f1a7f25699813785a222a0713c12fce1142.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8051/HighFidelity-Beta-PR12650-63e07f1a7f25699813785a222a0713c12fce1142.exe

File: libraries/networking/src/MessagesClient.h [hunk 1]
```diff
@@ -23,17 +23,137 @@
...
     MessagesClient();
     
     void startThread();
 
+    /**jsdoc
+     * Send a text message on a channel.
+     * @function Messages.sendMessage
+     * @param {string} channel - The channel to send the message on.
+     * @param {string} message - The message to send.
+     * @param {boolean} [localOnly=false] - If <code>false</code> then the message is sent to all Interface, client entity,
```
Comment (MiladNazeri) [inline libraries/networking/src/MessagesClient.h]: I think you left out @param {string} sender

File: libraries/networking/src/MessagesClient.h [hunk 2]
```diff
@@ -23,17 +23,137 @@
...
+     */
     Q_INVOKABLE void sendLocalMessage(QString channel, QString message);
+
+    /**jsdoc
+     * Send a data message on a channel.
+     * @function Messages.sendData
+     * @param {string} channel - The channel to send the data on.
+     * @param {object} data - The data to send. The data is handled as a byte stream, for example as may be provided via a 
+     *     JavaScript <code>Int8Array</code> object.
+     * @param {boolean} [localOnly=false] - If <code>false</code> then the message is sent to all Interface, client entity,
```
Comment (MiladNazeri) [inline libraries/networking/src/MessagesClient.h]: I think sender is left out here as well

File: libraries/networking/src/MessagesClient.h (hunk 1 above)
Comment (ctrlaltdavid) [reply libraries/networking/src/MessagesClient.h]: No, sendMessage() doesn't have a "sender" parameter. The messageReceived() function - used in the example - does, though, and this is documented in messageReceived().

File: libraries/networking/src/MessagesClient.h (hunk 2 above)
Comment (ctrlaltdavid) [reply libraries/networking/src/MessagesClient.h]: Ditto for sendData().

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Received is now correct.

**Pull request merged**